community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java

```diff
@@ -415,6 +428,30 @@ private static TemporalType find( int temporalType )
         }
     }
 
+    /**
+     * We need to range check nano ourselves, since Java allows a wraparound
```

Comment:
This does not feel as the right level to put this on. I think that the validation should go into the individual methods that takes nanoseconds, e.g. `org.neo4j.values.storable.LocalTimeValue#localTime(long)`.

Reply:
I can do this for the scalar types, but unfortunately not for the Arrays, since a `LocalTimeArray` is an array of `LocalTime` and not of `LocalTimeValue`.

Reply:
Actually, now that I tried implementing this, I think it lives in the right place. In general, it is OK for a `LocalTimeValue` to accept a nanosecond to be higher than 999_999_999 and wrap it around in the same manner Java allows that.

It is only in the property store, where we _know_ how we saved a value, and that nay nanosecond higher than that must have been a corruption.